pokemon-react-app/src/components/PokemonList.js

```diff
@@ -0,0 +1,53 @@
+import React, { useEffect } from 'react'
+import { useSelector, useDispatch } from 'react-redux'
+import axios from 'axios'
+import { setPokemons, setPokemonsLoading } from '../redux/actions/pokemonActions'
+import PokemonComponentList from './PokemonComponentList'
+import { Container } from '../styles/Container.style'
+import InputComponent from './InputComponent'
+import { Label } from '../styles/Label.style'
+
+function PokemonList() {
+    const { pokemons, loading } = useSelector(state => state.pokemons)
+    const dispatch = useDispatch();
+
+    const fetchPokemons = async () => {
+
+        dispatch(setPokemonsLoading())
+
+        const gen1Species = await axios.get('https://pokeapi.co/api/v2/generation/1')
```

Comment:
This first call is enough. You have here under the key `pokemon_species` all the info to fetch details only about 1st gen pokemons, so the second call (allPokemons) that is making now 1118 API call ca be reduced to only 151.

I appreciate you have used axios.all so they are called in parallel, but still 1118 calls is so much and makes the app fill slow when starting.